### DISCUSSION TITLE
Use TimeSeries for 7 day analytics data?

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#16168] [Q&A] [answered]
I was thinking about using TimeSeries to save som additional data to a document. In this case visitor data.
The data looks like this and is for a single document. it is refreshed every hour or so.
```json
[
        {
            "PageViews": 5,
            "Visits": 1,
            "Date": "2023-04-03T00:00:00.0000000",
            "PathName": "/"
        },
        {
            "PageViews": 15,
            "Visits": 8,
            "Date": "2023-04-01T00:00:00.0000000",
            "PathName": "/"
        },
        {
            "PageViews": 13,
            "Visits": 6,
            "Date": "2023-04-02T00:00:00.0000000",
            "PathName": "/"
        },
        {
            "PageViews": 290,
            "Visits": 16,
            "Date": "2023-03-31T00:00:00.0000000",
            "PathName": "/"
        },
        {
            "PageViews": 136,
            "Visits": 10,
            "Date": "2023-03-29T00:00:00.0000000",
            "PathName": "/"
        },
        {
            "PageViews": 93,
            "Visits": 7,
            "Date": "2023-03-28T00:00:00.0000000",
            "PathName": "/"
        },
        {
            "PageViews": 94,
            "Visits": 18,
            "Date": "2023-03-30T00:00:00.0000000",
            "PathName": "/"
        }
    ],
```

Firstly, is this a good match for TimeSeries?

If the answer is yes, should I use the `Append` method if I would like to replace the data every hour? I can't find a `c#` example in the docs so how would you do this?

**Comment by ayende** [comment, marked as answer]
Hi,
This is likely a great use case for incremental time series. You can increment a value for a particular timestamp (including concurrently, etc).
Without really needing to think about it. See:

https://ayende.com/blog/195268-C/ravendb-5-3-new-features-incremental-time-series-implementing-lambda-based-accounting
